ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,201 @@
+>
+```
+
+We then define the following Solidity interface:
+
+```solidity
+
+    struct PaymasterConfiguration {
+        address paymaster;
+        bytes paymasterData;
+        uint256 paymasterValidationGasLimit;
+        uint256 paymasterPostOpGasLimit;
+    }
+
+interface IPaymasterConfigurationResolver {
+    function getPaymasterConfiguration(bytes operation, bytes context) external;
+}
+
+```
+
+The wallet MUST perform an ABI-encoding of the entire `operation` object and make a view call
+to the `getPaymasterConfiguration` function of the `target` address.
+
+If the view call fails or returns an invalid data, the `wallet_sendCalls` method must fail and return with an error.
+
+### Validity Time Range Capability
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    validAfter: `0x${string}`, // operation valid only after this timestamp, in seconds
+    validUntil: `0x${string}`  // operation valid only before this timestamp, in seconds
+  }
+>
+```
+
+The wallet must then verify the time range [`validAfter`..`validUntil`] is valid and present it to the
+user in a human-readable way for confirmation as part of the transaction information.
+
+### Multidimensional Nonce Capability
+
+Identifier:
+
+`multiDimensionalNonce`
+
+Interface:
+
+```typescript
+type MultiDimensionalNonceCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    nonceKey: `0x${string}`,
+    nonceSequence: `0x${string}`
+  }
+>
+```
+
+For Smart Contract Accounts that support multidimensional nonce values,
+the wallet must specify these parameters during the actual on-chain execution of the batch.
+
+### Account Abstraction Gas Parameters Override Capability
+
+Identifier:
+
+`accountAbstractionGasParamsOverride`
+
+Interface:
+
+```typescript
+type AAGasParamsOverrideCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    preVerificationGas?: `0x${string}`,
+    verificationGasLimit?: `0x${string}`,
+    callGasLimit?: `0x${string}`,
+    paymasterVerificationGasLimit?: `0x${string}`,
+    paymasterPostOpGasLimit?: `0x${string}`,
+    maxFeePerGas?: `0x${string}`,
+    maxPriorityFeePerGas?: `0x${string}`
+  }
+>
+```
+
+Notice that all fields in the `AAGasParamsOverrideCapabilityParams` are optional.
+Only the values that callers want to override must be provided.
+
+In case `paymasterVerificationGasLimit` or `paymasterPostOpGasLimit` are provided,
+wallets should warn the users about this happening but use these values instead of
+the ones generated by any other capability.
+Wallets may choose to reject such configurations or request the user input in this case.
+Such a combination of features is only expected to be used in development and is very risky to use in production.
+
+### Set Externally Owned Account Code Capability
+
+This capability is designed to use with [EIP-7702](./eip-7702) transactions.
+
+Identifier:
+
+`setCodeForEOA`
```

Comment:
i don't think we need this at all - this is exactly what the wallet should do, and not the application. There is no way the wallet can protect the app from a malicious contract.

a wallet defines an account (address).
when the app performs a "sendCall", the wallet should decide how to implement it. if the network supports 7702, it should  submit it as a 7702 transaction. if that account already has a 7702 contract deployed (in a previous transaction), then just submit a batch call to use it.